### PR TITLE
Spelling: docs and comments

### PR DESCRIPTION
**Pull request by jsoref**
This PR corrects misspellings identified by the [check-spelling action](https://github.com/marketplace/actions/check-spelling).

The misspellings have been reported at https://github.com/jsoref/cvmfs/commit/b81da4ee22ded01ed33ed3b67b734db571249b84#commitcomment-61970638

The action reports that the changes in this PR would make it happy: https://github.com/jsoref/cvmfs/commit/6911cc0f6680169c96d11526d223774225437b82

Note: this PR does not include the action. If you're interested in running a spell check on every PR and push, that can be offered separately.

---

If you need things split, dropped, or squashed, just let me know, it isn't a big deal.

Note that one typo was identified by this tool and I sent a PR to remove it entirely.

I've tried to preserve whitespace indentation (including some trailing whitespace, which my editors hate).

I will try to call out a bunch of things (it'll take time).

### CODE REVIEW

File: cloc-1.64.pl [hunk 1]
```diff
@@ -69,16 +69,16 @@ BEGIN
     }
 }
 
-my $HAVE_Algorith_Diff = 0;
+my $HAVE_Algorithm_Diff = 0;
```
Comment (jsoref) [inline cloc-1.64.pl]: this might be a third party file that shouldn't be touched. let me know and i'll drop it.

File: cloc-1.64.pl [hunk 2]
```diff
@@ -537,7 +537,7 @@ BEGIN
    "lang_no_ext|lang-no-ext=s"               => \$opt_lang_no_ext         ,
    "yaml"                                    => \$opt_yaml                ,
    "csv"                                     => \$opt_csv                 ,
-   "csv_delimeter|csv-delimiter=s"           => \$opt_csv_delimiter       ,
+   "csv_delimiter|csv-delimiter=s"           => \$opt_csv_delimiter       ,
```
Comment (jsoref) [inline cloc-1.64.pl]: This is a bit odd, there are two divergences between the first and second thing, the `_`/`-` and the `e`/`i`

File: cloc-1.64.pl [hunk 3]
```diff
@@ -8225,7 +8225,7 @@ sub Get
 EOAlgDiff
 # 2}}}
     my $problems        = 0;
-    $HAVE_Algorith_Diff = 0;
+    $HAVE_Algorithm_Diff = 0;
```
Comment (jsoref) [inline cloc-1.64.pl]: the column alignment should probably be fixed

File: cpplint.py [hunk 4]
```diff
@@ -4716,22 +4716,22 @@ def _GetTextInside(text, start_pattern):
...
       return None
     elif text[position] in matching_punctuation:
       punctuation_stack.append(matching_punctuation[text[position]])
     position += 1
   if punctuation_stack:
-    # Opening punctuations left without matching close-punctuations.
+    # Opening punctuation left without matching close-punctuation.
     return None
-  # punctuations match.
+  # punctuation matches.
```
Comment (jsoref) [inline cpplint.py:4734]: this pair of changes is a bit different from everything else.

File: cvmfs/cache.h [hunk 5]
```diff
@@ -226,7 +226,7 @@ class CacheManager : SingleCopy {
  protected:
   CacheManager();
 
-  // Unless overwritten, Saving/Restoring states will crash the Fuse module
+  // Unless overridden, Saving/Restoring states will crash the Fuse module
```
Comment (jsoref) [inline cvmfs/cache.h:229]: overwrite and override are both words, and sometimes you overwrite something to override it, but I think in this case it's just the latter.

File: cvmfs/cache_extern.cc [hunk 6]
```diff
@@ -753,18 +753,18 @@ bool ExternalCacheManager::SpawnPlugin(const vector<string> &cmd_line) {
   int fd_null_read = open("/dev/null", O_RDONLY);
   int fd_null_write = open("/dev/null", O_WRONLY);
   assert((fd_null_read >= 0) && (fd_null_write >= 0));
-  map<int, int> map_fildes;
-  map_fildes[fd_null_read] = 0;
-  map_fildes[fd_null_write] = 1;
-  map_fildes[fd_null_write] = 2;
+  map<int, int> map_filedes;
```
Comment (jsoref) [inline cvmfs/cache_extern.cc]: both `map_fildes` and `map_filedes` were in use

File: cvmfs/cache_plugin/libcvmfs_cache.h [hunk 7]
```diff
@@ -224,7 +224,7 @@ void cvmcache_wait_for(struct cvmcache_context *ctx);
 uint32_t cvmcache_max_object_size(struct cvmcache_context *ctx);
 
 /**
- * Can be used to spawn a second process that superwises the cache plugin.
+ * Can be used to spawn a second process that supervises the cache plugin.
```
Comment (jsoref) [inline cvmfs/cache_plugin/libcvmfs_cache.h:227]: ?

File: cvmfs/cache_posix.cc [hunk 8]
```diff
@@ -172,7 +172,7 @@ int PosixCacheManager::CommitTxn(void *txn) {
                transaction->id.ToString().c_str(),
                transaction->expected_size, transaction->size);
       CopyPath2Path(transaction->tmp_path,
-                    cache_path_ + "/quarantaine/" + transaction->id.ToString());
+                    cache_path_ + "/quarantine/" + transaction->id.ToString());
```
Comment (jsoref) [inline cvmfs/cache_posix.cc]: api?

File: cvmfs/catalog_mgr.h [hunk 9]
```diff
@@ -191,7 +191,7 @@ class AbstractCatalogManager : public SingleCopy {
   }
   inline CatalogT* GetRootCatalog() const { return catalogs_.front(); }
   /**
-   * Inodes are ambiquitous under some circumstances, to prevent problems
+   * Inodes are ambiguous under some circumstances, to prevent problems
```
Comment (jsoref) [inline cvmfs/catalog_mgr.h]: The tool I used to pick a replacement suggested `ubiquitous` -- I sympathize, but I think this is the right word...

File: cvmfs/catalog_sql.h [hunk 10]
```diff
@@ -298,7 +298,7 @@ class SqlLookup : public SqlDirent {
                            const bool expand_symlink = true) const;
 
   /**
-   * DirectoryEntrys do not contain their path hash.
+   * DirectoryEntries do not contain their path hash.
```
Comment (jsoref) [inline cvmfs/catalog_sql.h:301]: My tool wouldn't complain about `DirectoryEntry's`, `DirectoryEntry-s`, or `Directory Entries`...

But, I didn't see any sign of `DirectoryEntry` so...

File: cvmfs/cvmfs_config [hunk 11]
```diff
@@ -1410,12 +1410,12 @@ cvmfs_bugreport() {
   mkdir etc
   cp -r /etc/cvmfs etc/
 
-  echo "Gathering files in quarantaine"
+  echo "Gathering files in quarantine"
   for repo in `ls "$CVMFS_CACHE_BASE"`
   do
-    qdir="${CVMFS_CACHE_BASE}/${repo}/quarantaine"
+    qdir="${CVMFS_CACHE_BASE}/${repo}/quarantine"
```
Comment (jsoref) [inline cvmfs/cvmfs_config]: This seems to be the pair to the other earlier item

File: cvmfs/cvmfs_talk.cc [hunk 12]
```diff
@@ -143,7 +143,7 @@ bool SendCommand(const std::string &command, InstanceInfo instance_info) {
 static void Usage(const std::string &exe) {
   LogCvmfs(kLogCvmfs, kLogStdout,
     "Usage: %s [-i instance | -p socket] <command>                     \n"
-    "   By default, iteratate through all instances.                   \n"
+    "   By default, iterate through all instances.                     \n"
```
Comment (jsoref) [inline cvmfs/cvmfs_talk.cc]: User facing documentation

File: cvmfs/dns.h [hunk 13]
```diff
@@ -256,7 +256,7 @@ class Resolver : SingleCopy {
 
   /**
    * Limit number of resolved IP addresses.  If throttle_ is 0 it has no effect.
-   * Otherwise, if more than thottle_ IPs are registered for a host, only
+   * Otherwise, if more than throttle_ IPs are registered for a host, only
```
Comment (jsoref) [inline cvmfs/dns.h:259]: User facing documentation

File: cvmfs/server/cvmfs_server_migrate.sh [hunk 14]
```diff
@@ -31,9 +31,9 @@ _migrate_2_1_6() {
   local new_upstream="local,${storage_path}/data/txn,${storage_path}"
 
   echo "--> removing spooler pipes"
-  local pipe_pathes="${CVMFS_SPOOL_DIR}/paths"
+  local pipe_paths="${CVMFS_SPOOL_DIR}/paths"
```
Comment (jsoref) [inline cvmfs/server/cvmfs_server_migrate.sh]: I don't see any particular reason for this spelling

File: cvmfs/server/test_cvmfs_server_util.sh [hunk 15]
```diff
@@ -92,11 +92,11 @@ mock_repo_name=""
 echo "Checking validity of repository name '$mock_repo_name'"
 print_check $(is_valid_repo_name $mock_repo_name; echo $?) 1
 
-mock_repo_name="abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQERSTUVWXYZ0123456"
+mock_repo_name="abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456"
 echo "Checking validity of repository name '$mock_repo_name'"
 print_check $(is_valid_repo_name $mock_repo_name; echo $?) 0
 
-mock_repo_name="abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQERSTUVWXYZ01234567"
+mock_repo_name="abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ01234567"
```
Comment (jsoref) [inline cvmfs/server/test_cvmfs_server_util.sh]: quite puzzled about these

File: cvmfs/statistics.h [hunk 16]
```diff
@@ -218,7 +218,7 @@ class Recorder {
 
 
 /**
- * Writes to multiple recorders.  Recorders with coarsed-grained resolution and
+ * Writes to multiple recorders.  Recorders with coarse resolution and
```
Comment (jsoref) [inline cvmfs/statistics.h:221]: `coarse-grained` is a thing (~5 million hits in Google), but `coarse-grained resolution` is barely at a thing (at 10k hits).

?

File: ducc/cmd/convert.go [hunk 17]
```diff
@@ -27,7 +27,7 @@ var (
 
 func init() {
 	convertCmd.Flags().BoolVarP(&overwriteLayer, "overwrite-layers", "f", false, "overwrite the layer if they are already inside the CVMFS repository")
-	convertCmd.Flags().BoolVarP(&convertAgain, "convert-again", "g", false, "convert again images that are already successfull converted")
+	convertCmd.Flags().BoolVarP(&convertAgain, "convert-again", "g", false, "convert again images that are already successful converted")
```
Comment (jsoref) [inline ducc/cmd/convert.go]: This should actually be `successfully`

File: ducc/cmd/loop.go [hunk 18]
```diff
@@ -16,7 +16,7 @@ import (
 
 func init() {
 	loopCmd.Flags().BoolVarP(&overwriteLayer, "overwrite-layers", "f", false, "overwrite the layer if they are already inside the CVMFS repository")
-	loopCmd.Flags().BoolVarP(&convertAgain, "convert-again", "g", false, "convert again images that are already successfull converted")
+	loopCmd.Flags().BoolVarP(&convertAgain, "convert-again", "g", false, "convert again images that are already successful converted")
```
Comment (jsoref) [inline ducc/cmd/loop.go]: ibid

File: ducc/docker-api/util.go [hunk 19]
```diff
@@ -44,7 +44,7 @@ var thinImageVersion = "1.0"
 
 // m is the manifest of the original image
 // repoLocation is where inside the repo we saved the several layers
-// origin is an ecoding fo the original referencese and original registry
+// origin is an encoding of the original references and original registry
```
Comment (jsoref) [inline ducc/docker-api/util.go:47]: My tooling defaults to ignoring 2 character words. So the `fo` change is a manual one-off.

File: ducc/lib/conversion.go [hunk 20]
```diff
@@ -74,7 +74,7 @@ func ConvertWishFlat(wish WishFriendly) error {
 		l.Log().WithFields(log.Fields{
 			"image":                  inputImage.GetSimpleName(),
 			"public path":            completePubSymPath,
-			"err stats pubblic path": errPub,
+			"err stats public path": errPub,
```
Comment (jsoref) [inline ducc/lib/conversion.go]: The go compiler will hate me here, so this block needs to have its whitespace fixed

File: ducc/notification/notification_test.go [hunk 21]
```diff
@@ -54,8 +54,8 @@ func TestNotificationWeCanReadThem(t *testing.T) {
 	readed := make([]Notification, 0)
 	scanner := bufio.NewScanner(f)
 	for scanner.Scan() {
-		byt := scanner.Bytes()
-		nn, err := ParseNotification(byt)
+		bytes := scanner.Bytes()
+		nn, err := ParseNotification(bytes)
```
Comment (jsoref) [inline ducc/notification/notification_test.go]: `byte` may be a reserved word, but i see no reason not to use `bytes` here.

File: externals/sqlite3/src/sqlite3.h [hunk 22]
```diff
@@ -913,7 +913,7 @@ struct sqlite3_io_methods {
 ** in order for the database to be readable.  The fourth parameter to
 ** [sqlite3_file_control()] for this opcode should be a pointer to an integer.
 ** That integer is 0 to disable persistent WAL mode or 1 to enable persistent
-** WAL mode.  If the integer is -1, then it is overwritten with the current
+** WAL mode.  If the integer is -1, then it is overridden with the current
```
Comment (jsoref) [inline externals/sqlite3/src/sqlite3.h]: sqlite's an external library. And there's a branch w/ my changes for it anyway, so the sqlite stuff should really be dropped.

If the entire `externals` directory should be skipped, let me know, that's easy to do.

File: gateway/internal/gateway/backend/leasedb.go [hunk 23]
```diff
@@ -83,7 +83,7 @@ type LeaseDB interface {
 	GetRepositoryEnabled(ctx context.Context, repository string) bool
 }
 
-// OpenLeaseDB opens or creats a new LeaseDB object of the specified type
+// OpenLeaseDB opens or creates a new LeaseDB object of the specified type
```
Comment (jsoref) [inline gateway/internal/gateway/backend/leasedb.go:86]: Google Sheets favored `cheats` for this and another one later....

File: test/benchmarks/003-alice/ppbench/Config.C [hunk 24]
```diff
@@ -1314,28 +1314,28 @@ AliGenerator* GeneratorFactory(PprRun_t srun) {
         break;
       case kFlow_2_2000:
       {
-        comment = comment.Append(" Flow with dN/deta  = 2000, vn = 2%");
+        comment = comment.Append(" Flow with dN/delta  = 2000, vn = 2%");
```
Comment (jsoref) [inline test/benchmarks/003-alice/ppbench/Config.C]: It's also close to `data` ... that's the problem with things that aren't recognized words, they're often midway between multiple candidates.

File: test/common/mock_services/make_repo.py [hunk 25]
```diff
@@ -47,7 +47,7 @@ def PredictResults(self):
 		print "Prediction:"
 		print "   directories to be produced:  " , directories
 		print "   files to be produced:        " , files
-		print "   bytes to be written (aprox): " , bytes
+		print "   bytes to be written (approx): " , bytes
```
Comment (jsoref) [inline test/common/mock_services/make_repo.py]: apparently i should adjust the whitespace for this block

File: test/src/607-noapache/main [hunk 26]
```diff
@@ -21,7 +21,7 @@ cvmfs_run_test() {
   src_location=$2
   local repo_dir=/cvmfs/$CVMFS_TEST_REPO
 
-  local replica_name="${CVMFS_TEST_REPO}.replic"
+  local replica_name="${CVMFS_TEST_REPO}.replica"
```
Comment (jsoref) [inline test/src/607-noapache/main]: seemed harmless

File: test/src/601-shake128/main [hunk 27]
```diff
@@ -32,7 +32,7 @@ cvmfs_run_test() {
   echo "starting transaction to edit repository"
   start_transaction $CVMFS_TEST_REPO || return $?
 
-  echo "createing nested catalog structure in repository"
+  echo "creating nested catalog structure in repository"
```
Comment (jsoref) [inline test/src/601-shake128/main]: This was mapped to `cheating`...

**Comment by jblomer** [comment]
Thanks for the PR.  To make it mergable, it would need to contain only the patches for non-external files.

**Comment by radupopescu** [comment]
Thank you for the patch.

I've been going over the list of changes. There are some changes which could have a functional impact on the system (directory names, for example).

It would be easier to integrate this if the first PR would only contain changes to documentation and comments.

Thank you!

**Comment by jsoref** [comment]
I can do that, but typically that'd result in a squash. It's a bit too expensive for me to do it per commit. Maybe this weekend.

**Comment by jblomer** [comment]
Closing this for the time being.  Please feel free to reopen as set of more targeted PRs as discussed in the comments.

**Pull request closed without merging**